ls1intum/Artemis
# [Bugfix] Parsing Error for Student Questions leading to Bad Request

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de and locally -> Bug seems to be only in production!


### Motivation and Context
Due to a parsing error, no questions can be posted in exercises and lecture Q&A; Server returns Bad Request as described in the related issue.
Fixes #3350 

### Description

Changes were made for `StudentQuestion`s and `StudentQuestionAnswer`s:
The field author (incl. `authorities` and `guidedTourSettings` which seems to be a problem for the Jackson parser) is not deserialized anymore, but set at the server to the current user right before saving the `StudentQuestion` or `StudentQuestionAnswer`.
Also an issue in `StudentQuestionAnswerResource` was fixed, as the user input in a requestBody of a PUT request was directly written into the database. Now, we allow changes (via PUT) only for the values `answerText`, `verified`, `tutorApproved`.

### Steps for Testing
As the bug was only prevalent in production, it is most important to check whether no regressions are introduced by these changes, i.e. everything should work as before posting/editing with different authorities.

1. Login as student
2. Visit an exercise or lecture page
3. Post questions and answers -> no bad request errors in console or browser should appear
4. Try to edit your questions and answers -> no bad request errors in console or browser should appear, also nothing else should change (such as author, date)
5. Try to edit questions/answers from others which should not be possible
6. Leave on question to be edited (in text or by approval) from an instructor of that course
7. Login as instructor
8. Navigate to a question from a student
9. Edit the text of a student's question -> no bad request errors in console or client should appear, check if the author name changes, after edit -> it should not!
10. Approve an unapproved question > no bad request errors in console or client should appear, check if the author name changes, after edit -> it should not!
